lib/core/src/Cardano/Wallet.hs

```diff
@@ -1933,17 +1899,26 @@ balanceTransactionWithSelectionStrategy
                         defaultTransactionCtx
                         boringSkeleton
 
-            -- Workaround for a corner case failure in
-            -- prop_balanceTransactionBalanced where.
-            --
-            -- Seems to be related with the wallet selecting a change output
-            -- with a coin value just about the 9 byte limit (4294967296). I.e.
-            -- seems like a problem of coin selection.
             feePadding =
                 let LinearFee LinearFunction {slope = perByte} =
                         view (#txParameters . #getFeePolicy) pp
                     scriptIntegrityHashBytes = 32 + 2
-                    extraBytes = 4
+
+                    -- Add padding to allow change and fee values to increase.
+                    -- Out of caution, assume both fields can increase by the
+                    -- theoretical maximum of 8 bytes each. Any overestimation
+                    -- will be reduced by 'distributeSurplus'.
+                    --
+                    -- The only downside is that we require the user to afford
+                    -- the fee for the padding, even if it later will be
+                    -- optimized away.
+                    --
+                    -- Also cf. 'costOfIncreasingCoin'.
+                    --
+                    -- NOTE: This should ideally be a constant defined next to
+                    -- 'costOfIncreasingCoin', but because of the package split
+                    -- the overhead currently seems to big.
+                    extraBytes = 16
```

Comment:
🤔 Actually, coin-selection should already overestimate the fee for the change ada value.

I think we only need 8 bytes here then. While coin-selection on its own overestimates the fee for the fee, `balanceTransaction` replaces the estimation with the exact `fee0 = minfee partialTx`. Because there's no change in the partial tx the overestimation of change will not be cancelled out.

Reply:
Fixed.